alby.go

```diff
@@ -712,3 +698,25 @@ func (svc *AlbyOAuthService) CallbackHandler(c echo.Context) error {
 	sess.Save(c.Request(), c.Response())
 	return c.Redirect(302, "/")
 }
+
+func albyInvoiceToTransaction(invoice *AlbyInvoice) *Nip47Transaction {
+	description := invoice.Comment
+	if description == "" {
+		description = invoice.Memo
+	}
+
+	return &Nip47Transaction{
+		Type:            invoice.Type,
+		Invoice:         invoice.PaymentRequest,
+		Description:     description,
+		DescriptionHash: invoice.DescriptionHash,
+		Preimage:        invoice.Preimage,
```

Comment:
can we always return the preimage? or are there cases where the preimage should not be returned (e.g. when the payment is not settled.)

Reply:
fixed